tests/integration/source_code/test_jobs.py

```diff
@@ -47,6 +46,34 @@ def test_job_linter_no_problems(simple_ctx, ws, make_job):
     assert len(problems) == 0
 
 
+def test_job_task_linter_no_problems(simple_ctx, ws, make_job, make_random, make_cluster, make_notebook):
+    created_cluster = make_cluster(single_node=True)
+    entrypoint = WorkspacePath(ws, f"~/linter-{make_random(4)}").expanduser()
+    entrypoint.mkdir()
+
+    notebook = entrypoint / "notebook.ipynb"
+    task = jobs.Task(
+        task_key=make_random(4),
+        description=make_random(4),
+        existing_cluster_id=created_cluster.cluster_id,
+        notebook_task=jobs.NotebookTask(
+            notebook_path=str(
+                make_notebook(
+                    path=notebook,
+                    content=b"""import doesnotexist
+import greenlet""",
+                )
+            )
+        ),
+        timeout_seconds=0,
```

Comment:
Will this kill the job instantly?

Reply:
0 is no timeout.

Reply:
https://docs.databricks.com/en/workflows/jobs/settings.html
> To configure a maximum completion time for a job, enter the maximum duration in the Timeout field. If the job does not complete in this time, Databricks sets its status to “Timed Out” and the job is stopped.

Reads like the job will be stopped directly and status set to "Timed out"

Reply:
From the [code](https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/service/jobs.py#L563) it says 0 is no timeout, but I removed it anyway since its optional.